src/goto-symex/goto_symex_state.cpp

```diff
@@ -625,6 +625,52 @@ void goto_symex_statet::rename_address(
   }
 }
 
+static bool requires_renaming(const typet &type, const namespacet &ns)
+{
+  if(type.id() == ID_array)
+  {
+    const auto &array_type = to_array_type(type);
+    return requires_renaming(array_type.subtype(), ns) ||
+           !array_type.size().is_constant();
+  }
+  else if(
+    type.id() == ID_struct || type.id() == ID_union || type.id() == ID_class)
```

Comment:
I don't think `type.id() == ID_class` should ever evaluate to true. A `class` is represented as `ID_struct` with `ID_C_class` set to `true`. If I'm wrong in saying this then we've got a lot more work to do elsewhere.

Reply:
Yes, these should be cleaned out.

Reply:
#3635